### PR TITLE
4751 notify trending map

### LINKED ISSUE #4751: Notify the user when a map becomes popular
Add a new notification to the user when the user's map becomes popular.

What is the number of likes to promote a map to popular?

//  @saleiva

### PR DESCRIPTION
**Pull request by ethervoid**
Closes #4751 

Check daily the mapviews to find maps that are becoming trendy and notify to their owners.

### CODE REVIEW

**Comment by ethervoid** [comment]
Please @Kartones check this PR. Give special attention to the change to Static maps url helper that I have to change in order to get the url from rake tasks

File: lib/static_maps_url_helper.rb [hunk 1]
```diff
@@ -3,18 +3,21 @@ module Carto
...
 
     def url_for_static_map(request, visualization, map_width, map_height)
-      static_maps_base_url(request) + static_maps_image_url_fragment(visualization.id, map_width, map_height)
+      username = CartoDB.extract_subdomain(request)
+      request_protocol = request.protocol.sub('://','')
+      static_maps_base_url(username, request_protocol) + static_maps_image_url_fragment(visualization.id, map_width, map_height)
+    end
+
+    def url_for_static_map_without_request(username, request_protocol, visualization, map_width, map_height)
+      static_maps_base_url(username, request_protocol) + static_maps_image_url_fragment(visualization.id, map_width, map_height)
```
Comment (Kartones) [inline lib/static_maps_url_helper.rb]: Gosh, how much I miss method overloading as in a decent language :_(

File: lib/cartodb/stats/api_calls.rb [hunk 2]
```diff
@@ -120,14 +120,18 @@ def get_api_calls_from_redis_source(username, api_call_type, options = {})
...
         first_requested_day = date_from.strftime("%Y%m%d")
-        first_requested_index = calls_in_reverse_order.index { |day_and_count|
-          day_and_count[0] <= last_requested_day
-        }
-        calls_in_reverse_order = calls_in_reverse_order.drop(first_requested_index) if first_requested_index
-        last_requested_index = calls_in_reverse_order.index { |day_and_count|
-          day_and_count[0] <= first_requested_day
-        }
-        calls_in_reverse_order = calls_in_reverse_order.take(last_requested_index + 1) if last_requested_index
+        if check_available_values(calls_in_reverse_order, first_requested_day, last_requested_day)
```
Comment (Kartones) [inline lib/cartodb/stats/api_calls.rb]: I would add some tests for this part...

Comment (ethervoid) [inline lib/cartodb/stats/api_calls.rb]: Added a test for the new scenario

**Comment by Kartones** [comment]
:+1: <3 the trending tests :)

**Pull request merged**